addons/mail/static/src/core/common/message.js

```diff
@@ -496,6 +520,121 @@ export class Message extends Component {
+                return persona;
+            }
+        }
+        return author || authorCheck;
+    }
+
+    async fetchMessageSuggestedRecipients() {
+        const partners = await rpc("/mail/thread/recipients", {
+            thread_model: this.props.thread.model,
+            thread_id: this.props.thread.id,
+            message_id: this.message.id,
+        });
+        return partners.map((data) => data.id);
+    }
+
+    async onClickMessageForward() {
+        const email_from = this.message.author?.email ? this.message.author.email : this.message.email_from;
+        const [name, email] = parseEmail(email_from);
+        const datetimeFormatted = _t("%(date)s at %(time)s", {
+            date: this.message.datetime.toFormat("ccc, MMM d, yyyy"),
+            time: this.message.datetime.toFormat("hh:mm a"),
+        });
+
+        const body = renderToMarkup(
+            "mail.Message.bodyInForward",
+            {
+                body: getNonEditableMentions(this.message.body),
+                date: datetimeFormatted,
+                email,
+                message: this.message,
+                name: name || email,
+            }
+        );
+
+        const attachmentIds = this.message.attachment_ids.map((attachment) => attachment.id);
+        const newAttachmentIds = await this.env.services.orm.call(
+            "ir.attachment",
+            "copy",
+            [attachmentIds],
+            {
+                default: { res_model: "mail.compose.message", res_id: 0 },
+            }
+        );
+
+        const context = {
+            default_attachment_ids: newAttachmentIds,
+            default_body: body,
+            default_composition_mode: "comment",
+            default_in_forward_mode: true,
+        };
+        this.openFullComposer(_t("Forward message"), context);
+    }
+
+    async onClickMessageReplyAll() {
+        const recipientIds = await this.fetchMessageSuggestedRecipients();
+        const email_from = this.message.author?.email ? this.message.author.email : this.message.email_from;
+        const [name, email] = parseEmail(email_from);
+        const datetimeFormatted = _t("%(date)s at %(time)s", {
+            date: this.message.datetime.toFormat("ccc, MMM d, yyyy"),
+            time: this.message.datetime.toFormat("hh:mm a"),
+        });
+
+        const body = renderToMarkup(
+            "mail.Message.bodyInReply",
+            {
+                body: getNonEditableMentions(this.message.body),
+                date: datetimeFormatted,
+                email,
+                message: this.message,
+                name: name || email,
+            }
+        );
+
+        const context = {
+            default_body: body,
+            default_in_reply_mode: true,
+            default_partner_ids: recipientIds,
+        };
+        this.openFullComposer(_t("Reply All"), context);
+    }
+}
+
+function getNonEditableMentions(body) {
```

Comment:
Possibly move this function to utils/format to go with other similar body processing functions.